src/main.rs

```diff
@@ -1095,6 +1105,72 @@ fn zpool_set(log: &Logger, pool: &str, n: &str, v: &str) -> Result<()> {
     Ok(())
 }
 
+fn zpool_trim(log: &Logger, pool: &str) -> Result<()> {
+    if pool.contains('/') {
+        bail!("no / allowed here");
+    }
+
+    info!(log, "TRIM POOL: {}", pool);
+
+    let cmd = Command::new("/sbin/zpool")
+        .env_clear()
+        .arg("trim")
+        .arg(pool)
+        .output()?;
+
+    if !cmd.status.success() {
+        let errmsg = String::from_utf8_lossy(&cmd.stderr);
+        /*
+         * This is not considered fatal, the pool's backing device may not
+         * support trim/discard.
+         */
+        info!(log, "zpool trim {} failed: {}", pool, errmsg);
+    } else {
+        /*
+         * Wait until the trim is complete. Unfortunately there is no nice
+         * parsable output mechanism for this information. When trimming is in
+         * progress, the output contains a line such as:
+         *   c6t5d0  ONLINE .. (2% trimmed, started at Mon May 15 ...)
+         * and when finished:
+         *   c6t5d0  ONLINE .. (100% trimmed, completed at Mon May 15 ...)
+         */
+        loop {
+            let output = Command::new("/sbin/zpool")
+                .env_clear()
+                .arg("status")
+                .arg("-t")
+                .arg(pool)
+                .output()?;
+
+            if !output.status.success() {
+                let errmsg = String::from_utf8_lossy(&output.stderr);
+                bail!("zpool trim status {} failed: {}", pool, errmsg);
+            }
+
+            let stdout = String::from_utf8_lossy(&output.stdout);
+            let status: Vec<&str> = stdout
+                .lines()
+                .filter(|x| x.contains("trimmed, started"))
+                .map(|x| {
+                    let s = x.find("(").unwrap_or(0);
+                    let e = x.find(")").unwrap_or(x.len());
+                    &x[s..e]
+                })
+                .collect();
+
+            if status.len() != 1 {
```

Comment:
Should we maybe wait for `trimmed, completed` to actually show up, rather than just the absence of `trimmed, started`?

Reply:
Looking at the zpool source, there is no state between active and complete, so I think we're ok here.
If somehow somebody got in and suspended the trim while it was running we want to just move on and export the pool anyway.